engine/src/main/java/org/camunda/bpm/engine/impl/batch/AbstractBatchJobHandler.java

```diff
@@ -169,6 +169,11 @@ protected void createJobEntities(BatchEntity batch, T configuration, String depl
       ByteArrayEntity configurationEntity = saveConfiguration(byteArrayManager, jobConfiguration);
 
       JobEntity job = createBatchJob(batch, configurationEntity);
+
+      if (jobConfiguration.getIds() != null && jobConfiguration.getIds().size() == 1) {
```

Comment:
Something like this will be required to avoid populating the ids for entities that are not process instance:
```suggestion
      if (jobConfiguration.getIds() != null && jobConfiguration.getIds().size() == 1
          && !(this instanceof DecisionSetRemovalTimeJobHandler)
          && !(this instanceof DeleteHistoricDecisionInstancesJobHandler)
          && !(this instanceof SetJobRetriesJobHandler)
          && !(this instanceof SetExternalTaskRetriesJobHandler)
//          && !(this instanceof BatchSetRemovalTimeJobHandler)
          ) {
```